articles/group-operator/group.md

```diff
@@ -0,0 +1,151 @@
+  
+### Aggregation
+
+Aggregation is an operation that processes the data to give a computed result. Aggregation operations group values from multiple documents together and can perform a variety of operations on the grouped data to return a single result. It basically involves stages or pipelines through which data is processed to yield a combined result. 
+
+Pipelines are stages through which data is processed, more technically transformed into by the provided criteria. Each pipeline is independent and receives the data of the previous stage. Always remember that the first pipeline will have direct contact with the data itself, not its transformed version, and it can take advantage of indexes.
+
+### What is $group operator?
+
+As the name suggests, the `$group` operator groups similar data by some specified expression and combines it into a single result or document.
+
+Suppose there are 15 people in a database and they all have a similar hobby. if we want to count all those people who have a hobby in common, then `$group` operator is an elegant solution for such a task.
+
+Let’s take a look at its syntax:
+
+`{ $group: { _id: <expression>, <field1>: { <accumulator1> : <expression1> }, ... } }`
+
+The `_id` field in the output documents contains the distinct group by key. The output documents can also contain computed fields that hold the values of some accumulator expression grouped by the $group‘s `_id` field.
+
+Basically, the `_id` field here will take those fields by which you want to group the documents. It is mandatory to use `_id` here.
+
+The $group operator does not remove the data from the database. It just aggregates multiple documents into a single document based on the field passed to the `_id` field.
+
+### Important points to remember
+
+- It can't be used with the `LogReduce` operator.
+
+- When parsing and naming (aliasing) fields, avoid using the names of grouping functions or other operators as field names.
+
+- When using count, or any grouping function, remember to include the underscore before the field name (sort by _count).
+
+- Multiple aggregation functions can be on the same line but you can't include another function, such as a math function, on the same line of a query.
+
+Now, let’s look at an example.
+
+Suppose we have a collection of 3 people with their names, gender, age and hobbies:
+
+```JavaScript
+[
+    {
+        "_id" : ObjectId("5ffb0dd58591ec5a52d2afbd"),
+        "name" : "Harit Joshi",
+        "age" : 18,
+        "gender" : "Male",
+        "hobbies" : [
+                "sports",
+                "cooking",
+                "gaming"
+        ]
+    },
```

Comment:
@HARITJOSHI1 You misunderstood me, Instead of just writing the syntax below the prerequisite section, Write like this:

---

Now, let’s look at an example.

Let's create a database.

```
// Command to create Database
```

Now, let's insert these documents into the database in a collection called person.

```
// Command to insert the documents
```

Now that we have some documents in the collection, we can use the $group to find how many males and how many females are there.

Reply:
You don't have to explain any of those commands though. That's clear from the prerequisites section.

Reply:
Ohh okay I will update it accordingly.